livekit-plugins/livekit-plugins-google/livekit/plugins/google/stt.py

```
@@ -449,9 +469,17 @@ async def process_stream(stream):
                     await utils.aio.gracefully_cancel(
                         process_stream_task, wait_reconnect_task
                     )
+                    should_stop.set()
+            except DeadlineExceeded:
+                raise APITimeoutError()
+            except GoogleAPICallError as e:
+                raise APIStatusError(
+                    e.message,
+                    status_code=e.code or -1,
+                )
+            except Exception as e:
+                raise APIConnectionError() from e
             finally:
-                if not self._reconnect_event.is_set():
-                    break
```

@jayeshp19 `break` here has the unexpected behavior of swallowing the exception. when this is set, the `except` in the parent class does not catch exceptions raised here.

---

good catch! let's move this inside try block

---

why do we need this break?

---

This is because we have an infinite `while` loop, so we need to break if `wait_reconnect_task` is not  in done. This means other tasks are either finished or had exceptions, and we don't want to retry in that case.